api.go

```diff
@@ -287,8 +287,10 @@ func (r *Repository) UnmarshalJSON(data []byte) error {
 		return err
 	}
 
-	id, _ := strconv.ParseUint(repo.RawConfig["repoid"], 10, 32)
-	r.ID = uint32(id)
+	if v, ok := repo.RawConfig["repoid"]; ok {
```

Comment:
is this branch necessary? r.ID is 0 either way

Reply:
Yes. In tests we set `Repository.ID` directly. Without this branch it would get set to zero.